plugins/single_plugins/vswitch.cpp

```diff
@@ -197,14 +197,17 @@ class vswitch : public wf::plugin_interface_t
 
     wf::signal_connection_t on_set_workspace_request = [=] (wf::signal_data_t *data)
     {
+        auto ev = static_cast<wf::workspace_change_request_signal*>(data);
         if (is_active())
         {
-            return;
+            ev->carried_out = add_direction(ev->new_viewport.x -
```

Comment:
I think for the sake of being 100% correct we should use `get_current_workspace` instead of `old_viewport` here, otherwise lgtm